tests/Integration/Connection/ConnectionTest.php

```diff
@@ -205,7 +205,7 @@ public function testOpenConnectionHandlesTheMissingGreetingCorrectly() : void
             $connection->open();
             self::fail('Connection not established');
         } catch (CommunicationFailed $e) {
-            self::assertSame('Unable to read greeting', $e->getMessage());
+            self::assertStringContainsString('Unable to read greeting', $e->getMessage());
```

Comment:
1. Same as above, if there is more information, can we check it (not necessarily an exact match, but some regexp pattern)?
2. Since you are changing `'Unable to write/read` to `Error when reading/writing`, should this error message be changed as well?

Reply:
Done